uni/lib/redux/action_creators.dart

```diff
@@ -542,6 +542,20 @@ ThunkAction<AppState> setFilteredExams(
   };
 }
 
+ThunkAction<AppState> setHiddenExams(
+    Exam newHiddenExam, Completer<void> action) {
+  return (Store<AppState> store) async {
+    final List<String> hiddenExams =
+        await AppSharedPreferences.getHiddenExams();
+
+    final String id = newHiddenExam.id;
+    hiddenExams.contains(id) ? hiddenExams.remove(id) : hiddenExams.add(id);
+    store.dispatch(SetExamHidden(hiddenExams));
+    await AppSharedPreferences.saveHiddenExams(hiddenExams);
+    action.complete();
+  };
+}
+
```

Comment:
I think you can pass only the exam.id and not the all object.

Also, toogleHiddenExam is a clear name for me...

Reply:
Yes, I agree it would be better to pass only the id.